lib/BoundaryClickWatcher/index.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class BoundaryClickWatcher extends Component {
```

Comment:
Epic work here, I really like this component. Just out of curiosity, what was your thinking behind calling the `targetIsCounter` variable? Did you mean `targetIsContainer`?